src/exampleTypes/Geo.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import { observer } from 'mobx-react'
+import { Flex } from '../layout/Flex'
+import { exampleTypes } from 'contexture-client'
+import injectTreeNode from '../utils/injectTreeNode'
+import AsyncSelect from 'react-select/lib/Async'
+
+const customStyles = {
+  valueContainer: styles => ({
+    ...styles,
+    ...{
+      overflow: 'hidden',
+      textOverflow: 'ellipsis',
+    },
+  }),
+}
+
+const elementStyle = {
+  flex: 1,
+  marginBottom: '5px',
+}
+
+const operatorOptions = ['within', 'not within']
+
+let GeoComponent = injectTreeNode(
+  observer(({ tree, node, loadOptions, placeholder = 'Address ...' }) => (
+    <Flex style={{ flexFlow: 'column' }}>
+      <select
+        style={elementStyle}
+        value={node.operator}
+        onChange={e => tree.mutate(node.path, { operator: e.target.value })}
+      >
+        {operatorOptions.map(o => (
+          <option key={o} value={o}>
+            {o}
+          </option>
+        ))}
+      </select>
+      <div style={elementStyle}>
+        <input
```

Comment:
This is probably fine, but doesn't follow our current "theming" conventions of passing in the actual input components as props. I'm definitely interested in hearing any ideas you have on how to make that better - the current thinking is to create a ContextureTheme object which we inject with Provider and pull components off that way. Thought?

Reply:
I did the `select` and the `input` as props. 

How are we going to handle the AsyncSelect?